tests/test_runtime.py

```diff
@@ -0,0 +1,736 @@
+        IBMQ.enable_account(token)
+        dev = IBMQSamplerDevice(wires=2, backend="ibmq_qasm_simulator", shots=shots)
+
+        @qml.qnode(dev)
+        def circuit(theta, phi):
+            qml.RX(theta, wires=0)
+            qml.RX(phi, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        theta = 0.432
+        phi = 0.123
+
+        res = circuit(theta, phi)
+        expected = np.array([np.cos(theta), np.cos(theta) * np.cos(phi)])
+        assert np.allclose(res, expected, **tol)
+
+    @pytest.mark.parametrize("shots", [8000])
+    @pytest.mark.parametrize(
+        "kwargs",
+        [
+            {
+                "return_mitigation_overhead": True,
+                "run_config": {"seed_simulator": 42},
+                "skip_transpilation": False,
+                "transpile_config": {"approximation_degree": 1.0},
+                "use_measurement_mitigation": True,
+            }
+        ],
+    )
+    def test_kwargs_circuit(self, token, tol, shots, kwargs):
+        """Test executing a simple circuit submitted to IBMQ."""
+        IBMQ.enable_account(token)
+        dev = IBMQSamplerDevice(wires=2, backend="ibmq_qasm_simulator", shots=shots, **kwargs)
+
+        @qml.qnode(dev)
+        def circuit(theta, phi):
+            qml.RX(theta, wires=0)
+            qml.RX(phi, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        theta = 0.432
+        phi = 0.123
+
+        res = circuit(theta, phi)
+        expected = np.array([np.cos(theta), np.cos(theta) * np.cos(phi)])
+        assert np.allclose(res, expected, **tol)
+
+    @pytest.mark.parametrize("shots", [8000])
+    def test_batch_circuits(self, token, tol, shots):
+        IBMQ.enable_account(token)
+        dev = IBMQSamplerDevice(wires=2, backend="ibmq_qasm_simulator", shots=shots)
+
+        # Batch the input parameters
+        batch_dim = 3
+        a = np.linspace(0, 0.543, batch_dim)
+        b = np.linspace(0, 0.123, batch_dim)
+        c = np.linspace(0, 0.987, batch_dim)
+
+        @qml.batch_params
+        @qml.qnode(dev)
+        def circuit(x, y, z):
+            """Reference QNode"""
+            qml.PauliX(0)
+            qml.Hadamard(wires=0)
+            qml.Rot(x, y, z, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(circuit(a, b, c), np.cos(a) * np.sin(b), **tol)
+
+    def test_track_sampler(self, token):
+        """Test that the tracker works."""
+
+        IBMQ.enable_account(token)
+        dev = IBMQSamplerDevice(wires=1, backend="ibmq_qasm_simulator", shots=1)
+        dev.tracker.active = True
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.PauliX(wires=0)
+            return qml.probs(wires=0)
+
+        circuit()
+
+        assert len(dev.tracker.history) == 2
+
+
+class TestCustomVQE:
```

Comment:
It would definitely be great to:

* Separate this class into its own file;
* Add more unit tests: units should be separated more and auxiliary functions like `hamiltonian_to_list_string` should have unit test(s).